src/tests/stress/pedsim/Ped_util/Global_funs.enc

```diff
@@ -0,0 +1,230 @@
+      end
+
+    end
+    var xml_way = file.children_named("waypoint")
+    var waypoints = new [(int, int)](|xml_way|)
+    var waypoints_distance = new [int](|xml_way|)
+    var i = 0
+    do
+      for a <- xml_way do
+        waypoints(i) = match (a.attribute_value("x").to_int(), a.attribute_value("y").to_int()) with
+                         case (Just(x), Just(y)) =>
+                           (x, y)
+                         end
+                         case _ =>
+                           print("BADINT!\n")
+                           abort("parsing error")
+                           (0,0)
+                         end
+
+                       end
+        waypoints_distance(i) = yolo_int(a.attribute_value("r").to_int())
+        i = i + 1
+      end
+      var nr_agents = 0
+      for a <- file.children_named("agent") do
+        nr_agents = nr_agents + match a.attribute_value("n").to_int() with
+                                  case Just(nr) =>
+                                    nr
+                                  end
+                                  case _ =>
+                                    print("BADINT2!\n")
+                                    abort("parsing error")
+                                    0
+                                  end
+
+                                end
+      end
+      var agents = new [Agent](nr_agents)
+      var i = 0
+      do
+        for a <- file.children_named("agent") do
+          var max = yolo_int(a.attribute_value("n").to_int())
+          var x = yolo_int(a.attribute_value("xs").to_int())
+          var y = yolo_int(a.attribute_value("ys").to_int())
+          var dx_max = yolo_int(a.attribute_value("dx").to_int()) / 2
+          var dy_max = yolo_int(a.attribute_value("dy").to_int()) / 2
+          var agent_waypoints = new [(int, int)](|a.children_named("addway")|)
+          var dx = x - dx_max
+          var dy = y - dy_max
+          var j = 0
+          do
+            for b <- a.children_named("addway") do
+              agent_waypoints(j) = waypoints(yolo_int(yolo_string(b.attribute_value("id").substring(1, 2)).to_int()) - 1)
+              unless |agent_waypoints| > 1 then
+                print("THIS SHOULD NEVER HAPPEN |new_agent.waypoint|={}\n", |agent_waypoints|)
+                abort("parsing error")
+              end
+              j = j + 1
+            end
+            for unused <- [0..max-1] do
+              if dx > x + dx_max then
+                print("out of space\n")
+                abort("parsing error")
+              else
+                if dy > y + dy_max then
+                  dy = y - dy_max
+                  dx = dx + 1
+                else
+                  dy = dy + 1
+                end
+              end
+              agents(i) = new Agent((dx, dy), copy_tr(agent_waypoints), copy_ar(waypoints_distance), i, 10000)
+              i = i + 1
+            end
+          end
+        end
+        agents --consume
```

Comment:
agent is supposed to be linear, but can't be due to limitations of the current type system. In order to make this easier too change later, I added `-- consume` where it actually should say `consume(...)`